action/protocol/poll/governance_protocol.go

```diff
@@ -399,29 +378,41 @@ func (p *governanceChainCommitteeProtocol) readBlockProducersByEpoch(ctx context
 	updatedVotingPower := make(map[string]*big.Int)
 	intensityRate := float64(uint32(100)-unqualifiedList.IntensityRate) / float64(100)
 	for _, cand := range candidates {
-		candidatesMap[cand.Address] = cand
 		if _, ok := unqualifiedList.BlacklistInfos[cand.Address]; ok {
 			// if it is an unqualified delegate, multiply the voting power with kick-out intensity rate
 			votingPower := new(big.Float).SetInt(cand.Votes)
-			newVotingPower, _ := votingPower.Mul(votingPower, big.NewFloat(intensityRate)).Int(nil)
-			updatedVotingPower[cand.Address] = newVotingPower
-		} else {
+			cand.Votes, _ = votingPower.Mul(votingPower, big.NewFloat(intensityRate)).Int(nil)
+		}
+		if cand.Votes.Cmp(big.NewInt(0)) > 0 {
 			updatedVotingPower[cand.Address] = cand.Votes
 		}
+		candidatesMap[cand.Address] = cand
```

Comment:
if votes not larger than 0, shall we still add it to candidatesMap?

Reply:
true! fixed on latest commit